test/functional/p2p_blocksonly.py

```diff
@@ -55,8 +55,51 @@ def run_test(self):
         with self.nodes[0].assert_debug_log(['received getdata for: tx {} peer=1'.format(txid)]):
             self.nodes[0].sendrawtransaction(sigtx)
             self.nodes[0].p2p.wait_for_tx(txid)
+            pprint(self.nodes[0].p2p.wait_for_tx(txid))
             assert_equal(self.nodes[0].getmempoolinfo()['size'], 1)
 
+        self.log.info("Restarting node 0 with whitelist permission and blocksonly")
+        self.stop_node(0)
+        self.start_node(0, ["-whitelist=127.0.0.1", "-whitelistforcerelay", "-blocksonly"])
```

Comment:
I think if you add nopersistmempool, it will drop the tx from the mempool and you can reuse it for this test without having to construct it again. In fact the below construction looks similar to the one above, so the resulting tx might be the same, thus leading to test failures.

```suggestion
        self.start_node(0, ["-whitelist=127.0.0.1", "-whitelistforcerelay", "-blocksonly", "-nopersistmempool"])
```

Reply:
Thanks Marco, going to try again with the suggestion!